packages/hebao_v2/contracts/price/UniswapV2PriceOracle.sol

```diff
@@ -48,9 +49,9 @@ contract UniswapV2PriceOracle is PriceOracle
         }
 
         if (token < wethAddress) {
-            return amount.mul(reserve1) / reserve0;
+            return amount.mul(reserve1.mul(PRICE_SCALE) / reserve0) / PRICE_SCALE;
```

Comment:
Hmmm... Doesn't this behave the same as with without the price scale? (except that this will overflow earlier).

Reply:
How about this:
```
    if (amount > 2 ** 128) {
        return (amount / reserve0).mul(reserve1);
    } else {
        return amount.mul(reserve1) / reserve0; 
    }
```

Reply:
Is there a specific problem we're trying to solve with changing the math?  I don't think there's a realistic scenario where `amount > 2 ** 128` that will result in a valid transaction.

Reply:
see this tx: https://dashboard.tenderly.co/tx/mainnet/0x61e413faa4ae2a768d91742aab8106f0b4c424d72c913e792da83cd2befa2584

Reply:
But this is with quota enabled right? If so, a failing tx is unavoidable like I explained above: https://github.com/Loopring/protocols/pull/2581#issuecomment-986970889

You can fix the overflow but the tx would just fail at a later point. Infinite approvals cannot be used together with a daily quota. The wallet owner needs to disable the daily quota if he wants to use infinite approvals.

Reply:
Yes, I agree. For some reason, we use the uint.max for all approvals.